/*
http://adventofcode.com/2017/day/9
--- Day 9: Stream Processing ---
A large stream blocks your path. According to the locals, it's not safe to cross the stream at the moment because it's full of garbage. You look down at the stream; rather than water, you discover that it's a stream of characters.

You sit for a while and record part of the stream (your puzzle input). The characters represent groups - sequences that begin with { and end with }. Within a group, there are zero or more other things, separated by commas: either another group or garbage. Since groups can contain other groups, a } only closes the most-recently-opened unclosed group - that is, they are nestable. Your puzzle input represents a single, large group which itself contains many smaller ones.

Sometimes, instead of a group, you will find garbage. Garbage begins with < and ends with >. Between those angle brackets, almost any character can appear, including { and }. Within garbage, < has no special meaning.

In a futile attempt to clean up the garbage, some program has canceled some of the characters within it using !: inside garbage, any character that comes after ! should be ignored, including <, >, and even another !.

You don't see any characters that deviate from these rules. Outside garbage, you only find well-formed groups, and garbage always terminates according to the rules above.

Here are some self-contained pieces of garbage:

<>, empty garbage.
<random characters>, garbage containing random characters.
<<<<>, because the extra < are ignored.
<{!>}>, because the first > is canceled.
<!!>, because the second ! is canceled, allowing the > to terminate the garbage.
<!!!>>, because the second ! and the first > are canceled.
<{o"i!a,<{i<a>, which ends at the first >.
Here are some examples of whole streams and the number of groups they contain:

{}, 1 group.
{{{}}}, 3 groups.
{{},{}}, also 3 groups.
{{{},{},{{}}}}, 6 groups.
{<{},{},{{}}>}, 1 group (which itself contains garbage).
{<a>,<a>,<a>,<a>}, 1 group.
{{<a>},{<a>},{<a>},{<a>}}, 5 groups.
{{<!>},{<!>},{<!>},{<a>}}, 2 groups (since all but the last > are canceled).
Your goal is to find the total score for all groups in your input. Each group is assigned a score which is one more than the score of the group that immediately contains it. (The outermost group gets a score of 1.)

{}, score of 1.
{{{}}}, score of 1 + 2 + 3 = 6.
{{},{}}, score of 1 + 2 + 2 = 5.
{{{},{},{{}}}}, score of 1 + 2 + 3 + 3 + 3 + 4 = 16.
{<a>,<a>,<a>,<a>}, score of 1.
{{<ab>},{<ab>},{<ab>},{<ab>}}, score of 1 + 2 + 2 + 2 + 2 = 9.
{{<!!>},{<!!>},{<!!>},{<!!>}}, score of 1 + 2 + 2 + 2 + 2 = 9.
{{<a!>},{<a!>},{<a!>},{<ab>}}, score of 1 + 2 = 3.
*/
function groupSum(input) {
  var sum = 0;
  var depth = 0;
  var inGarbage = false;
  for (let i = 0; i < input.length; i++) {
    if (inGarbage) {
      if (input[i] == "!") {
        i++;
      } else if (input[i] == ">") {
        inGarbage = false;
      }
    } else { // !garbage
      if (input[i] == "<") {
        inGarbage = true;
      } else if (input[i] == "{") {
        depth++;
        sum += depth;
      } else if (input[i] == "}") {
        depth--;
      }
    }
  }
  return sum;
}
console.assert(groupSum("{}") == 1, "single");
console.assert(groupSum("{{{}}}") == 6, "triple-nested");
console.assert(groupSum("{{},{}}") == 5, "two double-nested");
console.assert(groupSum("{{{},{},{{}}}}") == 16, "two triple and a quad");
console.assert(groupSum("{<a>,<a>,<a>,<a>}") == 1, "single garbage");
console.assert(groupSum("{{<ab>},{<ab>},{<ab>},{<ab>}}") == 9, "four doubles");
console.assert(groupSum("{{<!!>},{<!!>},{<!!>},{<!!>}}") == 9, "four double !!");
console.assert(groupSum("{{<a!>},{<a!>},{<a!>},{<ab>}}") == 3, "one double garbage with !");

/*

What is the total score for all groups in your input?

http://adventofcode.com/2017/day/9/input
*/

const day9input = "{{{{{{<e<{!i!>},<oe'!><!}}\"ao,o>},{{},<!!!>e}\"!!!!!>!}!>,<\",!}!>},<{>}},{<{e!!i{\"!!!>!!!>>,{<!!e!>},<!>,<!>!u!!!!!!i{!><!>!!!>},<!!,!!!>!!,>}}},{{{{}},{<a!!!e<o!!!>!u,!!!>,<\"io!!{!>},<\"!!!>!!!>i>}},{{<eui!!!>e,e!>,<>},{}}},{{{{}},{}},{{{<{,i\"!>!!\",u{\"!>!>'!!u>}},{{}}},{{},<\"}'{!>,<{!!!>a\",!!!>>}},{{<i<ao\"'a!>!>},<!!ouo\"!!!!>,{{{<}!!!u\"u!!!!!!'<ui!!!>'!!i<!!!>ie\">},{{<e!!!>o{!>,<>},{<}!>,<<uo!>},<i,o!!\"a!}!>},<}!>},<>}}}}},{{{<!>'>},<!!>},{{<i>}},{{<{!!u>},<e'aa{'<!>,<<auo!{!!!!u>}},{<{!!\"!!!>!!!>!\"!>,<!>},<u!!!a>,{<\"!!!>'<!'!,\"}}!>},<i}i!!!><!!!>i>}}}},{{{{<!!!>\"}!>i!a{>},{{{{<!>},<o!!<o!u!>,<}!>,<a\"!!!>,<'!!>}},{<,\"!!\"uu<,<!>',i\"!>!>,<,!<>}}},{{{},{{<\"!!!{ai!!!>},<u!!!>,<\"e!>!>!>{<!!!!!>>},{<uu!!uo}'!>},<!>!u'a\"i<}!>,<'>,{<i!!!>},<!!!<!>},<>}}}},{{{<\"!>},<}\">,<!}!>!>},<>}},<'!>!!!>,<!'!!}!>ei!!!>,<!>,<<!>,<!>,<>},{{<!>,<e!\"<!!!!!>,<!>},<!!!!!'u!>,<!!!>uo!!!>}a>},{{<ae!!a\"i!>,<!>},<!>,<!!o'}>}}}}},{{<!>\"'ue!!!>!!>}}},{{{<'e!>,<!\"!!!>>}},{<!>,<!ea!!!>!o!!!!!!!>'>},{{{{<!!!!!!!>!>!>,<!>},<<,<{>}},<}ue!,{,!io>},{}}},{{{<'!i!!i!!!>,<\"!>!>},<>,<,!!,!!}{!>},<{e!>},<ii{!!!!!>>},{{{<,<ii,>}}}},{{{<{{e!!!>!<i!>!!!>u!!!!{{a!!!!{!>>}},{{},{}}},{{},{{<<!!!>!!!>{!!!>,<\"!!{<eo!}!!!>!!!>{>,{<i>}},{{{}},{<!e!>,<{!>!!!>i>}},{{<!>,<!>},<>},{{<!!!>i!>,<>}}}}}},{{{{{<>}},{{},{{<!!!>'!!>}},{<oa!!!>!}e>,{}}}},{{{<oe!!u!>a!!!>ii<{{!!!!!!!><,!>,>}},{<!!!!!i!!!>'e,'!>,<u!a!!\"!!!{,!{!>,<e!!!!!>},<}>},{{},{<}!!i!>,<a!>},<a!>,<!!\"o!!!!!>,<<!!>,{<o!}<!!\"{\"o}!>},<!!!><u!!}u!>},<!!!>!>,<\">}}}}},{{{{<!!i,!!{o!>a!!!>!>},<!u!!!>e{!!!>!!u!!!!a}>}},{<!!u!<\",a}!!!>{{}o!>},<e\">}},{<!!!>!>},<!!\"o!i!!}u!>},<{<!>},<!!!!!!>,{<!!e!!i,'<e\"o>,<!!\"<aeo>}},{{<'e}!>,<!!}!\"!!!<!!,!!!>,<!>,<'u!}!>},<{,>},{<>}}},{{<u!!}!!!>!!i!!a!>},<!>>},{},{{},<a!!!>e{!!!>!!!!!>i}e,}!!a!}!>},<>}}}},{{{<!}!>},<>,<'!!o>}},{{{{},{{<!>},{a!>},<>},{<i}u>,<,!>,<!>},<!!!>,!>!>},<{!!!>',ae!<!>},<!!!>e'>}}}},{{{{{{<o!<uoo!!!>}\"o,'{!,a,>}},{{<<!o!>,<',\"'!'!>,<!!!>}a<!!>}}},<!!u!!'e!<!>,<!!!a,u!>},<!!!>},<>},{},{{}}},{{{{{{{<<!!e!!<u<!!\"i!!>},{<i\"!>},<!!!>},<o{!>},<!>,<{{a!>},<i!{e>}},<\"!,!!u!!>},{{}},{{<u!>},<>},{<!,}a{oo<}{}i<!!!>eu\"!>,<>}}},{{<!>},<ae!>,<!>!>>,<!!!!uei{!!!!'}!o{u{!<!>,<,>},{{<'a!>,<!!o,!>},<!>!>},<!>},<<!!\"!!>},{}},{{{<,!\">}},<!>,<!!aa,!>},<e!!!!!>},<!>{!!!>e'!!i!!!>,<>}}},{{{{<i!!!!<u'!>o!>,<!!!>},<\"\"!>,<!\"ui!i'}!!!>>},{{<,ao!>},<!>,<!}e!!!!i{a!>,<!u<!'}<o>}}}},{{{<!,}{{{!>},<!!!!,uia!a<}ei\"!!>,{}},{<i!!}!>},<!>,<'!<ou>}},{<oo}a{!o!!<!>},<!!!>!!!!!>!!e!{!!u!>},<,!!e!e!!,>,<!!!>,<\"<'!!!>},<>}},{{<!!{>},{}}}},{{{<!'!>,<!!{!>},<!>,<o<!>,<!>,<>}},{<!!!i}!u'!>,<{u!!ea!,a'o!!!!!!!>i!{>}},{{<a},i!!!!!>,<\"<<!!!!!>{>},{{}},{{<!!!!\"'u{!!!o!!!>},<!!u>},{<u!!o!>},<'!>,<o}!>,<u!!<!>,<!!!>>,<i\"'i{!>,<e\"'!}>}}}}}},{{{<'!u,{a!!!>u!>},<!>,<e>}},{{<}!>},<!{!!!>},<!!!>a!'!!!>!o!!,<o>,{<{<!!!>!!!>!!!!<}\"!!!!>}},{{}},{{<!>',!!i!!u\">,{<o,!!!>a!>},<a!!!!!>!!,ai,!!>}},{{<'!!!>!{!!!>},<{\"\",!!!>!>},<o<!!!>!!{>},{}}}}},{{{{}},{{{<!>},<!>},<!!!>ii!!!>!!!>,<i<!!'<{>},{<}!>>}},{{<e!>},<!{e!>!!!>{>,<>},{{{<,!>},<e<a\"o!!!!!>,}''>},{}},{},{<{'>,<!!!>{\"!>,<u>}},{{{<{\"eu!a!>,<!>e!u,i\"u!!!>,<}!><>},{{<!!!>,<!!!>!!\"i!!\"}!>},<\"<!!o,!\"!eo!!!>!!!>i!i>}}},{<!e>,{<\"e!>,<!!{!><<,u>}}}},{<>}},{{<<e<!!a>}}},{{{{<u''{!>!o'>}},{{{<>}},{}},{{<'!>},<uia!>},<!!!!!!u!>!>!uu!>,<!!!>>},{{<<'i!!!>,<!!!>,<},\">},{<!>o{aiae}u}e>}}}},{{{<!!u,<uu},i'!!!!!>!>,<i>},{<!>,<!\"!,}a!!ii{<<{{!>>,{{{<!!!><,!!''ie<!>},<!u!,aa>},{}}}}},{<e!!\"!>},<!!!!!>}!!!>o,!!iu{ua!>,<!!!!!!a>,{<io'!!e!!!!!>!>},<e,<o>}},{{<oi>},<}o!!!>!,!!!>!!u>}},{{{{{{<!!}!>,<!>,<!>>},{<,>}},{{{},{{<{>},{<{!>},<'!e}!a!!!>o'>,<e!{,a!!!}!!i!!<{'!>!!<\"!>>}}}},{{{<>}},{{<u!>!>,<!!!>},<!>},<io!!''io!>,<>}}}},{{{<,a<o!>},<!!,,}o!!!><\"!!,i!!!>,>},{{{},{}},{}}}},{{},{<!>e'a!!!><u{!>\"!}>,{<>}}}},{{{{<!>},<<!!!>oe'!}!>},<'au,oai!>!{!>,<>}},{<\"a!!ee!>,<!!!!a}!>\"!u\"{',>,<!!!>\"!>!!a{'!>,<\">}},{{{{{{<!>,<u!e!>ae!!!>!>},<{!!a!>},<>}},{<a{!>,<'i!!o}}!!,i!>,<!!{\"!e!!\"!ou>}},{{<!!'!!!<a!'<!!!>\",>}}},{{{{{<!>,<!,a!!!>,<!>,<>}},{}}},{{<!!!!}i!!!>,<!!!!{'euu>},{{<}>}},{{{{{},<!,o,!'!>},<}aie!!!!!>!!!>!!!>},<!<{{!!!>,<!!!><>},{{<o!>},<!!!>!!!>,!!!a!!{{a{!!<a\"i>}}},{<!>,<'!!!><ai!{!!<e\"a<'!!{!'!!u!>,<a!!!>>}},{{<!!!>o,,>}}}},{{<!!i'!!{>,<\"!>,<!!,<oi\"'iu!>ou>},{{<!>},<>},<!\"o!!,u,!!!>!!!>{,,}'<!!!>e!>},<<>}},{{{{<'!>,<{!!!>o!!u,u>}},{},{{{{{<!!>,{<}e',!!!!!>}!>,<{!!!>!!!>a'uia>}}},{{{{<!!,!!!!!>!>!!i>},{{<\"!>},<{!!'!>},<!!!e!!!!!>}!>>}}},{{},<a'!!!!\"!>,<!>,<{!>,<a,!>,<,!!oaa>}},{}}}},<{u!!!>{'!{}!!u!!!>},<!!}>}},{{<!!o!!!!a!>},<!>},<ie\"{\"i>,{<!!!>}'a!>uii{o\">}},{{<!>,a'!!'ea<'!!e'{\"u!>,<{,!!}>},{<!!!!!>,<!!!>!>},<{,<>}},{<{!>},<!>},<!!!>}>}},{{{<!!!><!o!>!i!>,<!>!>o!!!>!!!!!!,>,{{<\"uu!,a<ii'{u{!>},<{<!!,!<>}}}},{<{'!>!!\"i>,{<!>},<!{!!!!u\"<i!!!>ie>}},{}}}},{{<}!!,,e!!!!!><\",<!ia>}}},{{{<o!!!><>}},{{}}}},{{{<a!>iu'\"e>}},{{},<,\"'!!!!!u>},{<!>,<!!!>\"e!>,<!!}!!!>!!o{'a>}}},{{{<!!!a,!>,<!}!!o'!!e>},{<<<e\">,{{<!!!>!!!>!,!>},<!!'}!!!!!>!>,<!!!>},<a>},{<a!!\"<!!!{a!o!!!>!i}i>,<!eu!!u!i\"!!!!}o!!!!!!>}}}}},{{<!!o!>},<i\"'e!!<{!!!>>}}},{{<!!u\"<!,!!!\"oie>,<i,eo<>},{},{<u{!!!>,<a>,{{<!>,<!>,<!!!iu!!!>!>,<!!ao!!!>!i\"!{!>>}}}},{},{{},{{<!>},<!e>,<!ie!>},<!!i!>uio!>!!!>ui>},{},{<!'!e!!a'!>},<ii>,{<{,>}}}}},{{{{<ue!>,<}oa,}!!>},{<,!!!!!>i,e,!>},<!\"<!a}i<a>}},{<\"!>,<!!a!>,<,>},{<,!>},<\"o!!i!!!>!>!!!>ui{!!!>a'i!!!>},<{!!>}},{{{}},{{<!u!>a'!>!!o<!!,,o<!!o>}},{}},{}}},{{<!>!!!>!>,<\"'}{!>,<!!!>>},{{<'{!'o!>,<o\"\"e!!,o!!!>,<e!!!>!>o>},{<!!!!a}'!!e!>},<!>},<!!e'>}}}}}},{{{{{<{u!!!>aeu'>},<!!!>,!>!i!!a\"o!>!u!!\"!>},<o!,{!>,!!',>},{<!!!>o\",}o!>!!!>'!!o'!!e!,<{!!u>,{<!!ou!>!!o,\"!!!>\"!u,!!{e,}o>}},{{{{<'!!!>},<>},<!!{'!>},<\"!!<a<!>},<i!!a!>,<>},{}},{{<!e<e!!{>},<{!>,}!!a,>},{{<o!!,>},{}}}},{{{<o\"!>,<u!>u!>},<!!!>!>ee>},{},{{{<!!!>{!\"\"!>},<a>},{}},{{<au,!>'>}}}}},{{{{<'!!e,u!i'',!>,<!><,!>,<\"!>,<>}},{{<!>,<!'!>,<!>>}},{{},{{<!!i!>},<\"!<!!ue!>},<e}\"oee!!<!>},<>}}}}},{{{{{<!>,<<!!!>!!!>u!>!>,<!!i}<i,\">},<a{!!\"!>!>,<uo!!,!!!!>}},{{{<e\"!!a!!}'{<>},{{<}i'<a!u}!e!!!>!>,<!!!>,!\"!>,<!>,<>}}},{{{},<e!!{'o!!!>u!>,<!!!>'!>},<}!>,<}!>!!!>>},{<!!u!!!!!>!>,<!!<},!!,'o<!!!,,<!>}\">},{{<e!!!>\"<{!!i'\",ae'!!!>i>},<o!!!!!>{<aeu<!'!>,<>}}}},{{{{<>}}},{<uieoou'}<i!!!>!!!>!>>,{}},{{<o\"e'<\"!!e,!'e!>},<\">}}},{{{{<!i'!!i'!!e!}e!a!>},<e!>,<o!!>},{{<}!\"!!i{!!u}!>},<u\"u<'e}>},{<,i!>},<!>!>,<{{,!>!!!>!!!>\"''i!!!>,<!<!!<>}}},{{{<oe}!!\"!!!>,<\"eo,>},{<'!o<,}>}},{<!{!>,<!>,<'<!!}e'!<!\"a{e!>o'!>,<'!!!>>}},{<!oa{!!'!,a!!,u!>!!a,!>e!!!>\",oi>}},{{{{{{<e}!>,<<e}!!oa>}},{{<'<!!!>,<\"!>,<>},{{<!><!>'u>}}}},{{{{{},<\"!!i!{!>!>,<{i!!!'!<ei!>!>},<!!!!e!>},<{a!!>},{{<e>},{<'uiaoe!!,!!o!!\">}}},{{},{}}}}},{{<a}!>},<!>oo!>},<!iu!>},<>,<!!aoae}>}},{}},{{<!i}{a!>,<{}!>},<<a>,{{<\"!>!!!!au!>,<!!!!!!u\"<>},{{{<!!{{\"u}!!}a!>,<!'!>,<}}a>}},<o!!!!u!>},<',o<i{!>},<!!a}>}}},{{{{<oi!'o!>,<ao<ouie!!!>{'\">},{}}},{{<!e!>,<>},{<e!>},<!!}ei{}o!!!u,!>}!},!>,<>}},{{{<!>{!,o<aa!}'!!!o'!!!>!!!!!>>},{}}}}}},{{{},<!\"}!>!!!!\"!!!>a!!!>!!!>>},{{<\"eoe!!!>>,{<u{'io!>},<!!\"!>>}},<!>!!',,'!>eui<!e<i,!!!!!''!!{<\">}},{{{{},{<i'o!>},<'<a!i{o>}},{{<u\"!!<\"!!!!!>!><au!>},<!!!!!>'a}'!u!>,<>},<}'}!!!>!o,!>,<!>},<!!a!!!>ea>}},{}}}}},{{{<e<!!!>\"!!!>>,{{<!>!>!>},<!>,<!>!>!>,<!>},<!>'i!>o>}}}},{{{},{<!!!!!>},<!>},<a!>,<!!}!\">,{{},{<,!>!!!!<!!u>}}}},{{{{{<!!{!!!>i>},<o''!\"'e!!!{,a!<a!!!>,>},{},{{<!>,<{!>},<{ua>},{<!>'!!!!!>,<>},{{{<{o\"!>},<!!o{!!{e<}!!{'!>},<>},{<i!}u!!!>\">}}}}},{<!!!>,<!>,<,<!>>},{{{<!>},<>},<<!!u{!>!!!>},<!>,<!>!e!!i<!>,<!!i\"!!!u{}!>},<!!u>},<'!!e}!!!>,!>}>}},{{<e!>,<!>!,!!{!!!>!!!!!>}!>>}},{{{<\"eiuu!>>,{{}}},{{{},{<!!!>i!\"{>}}},{<!!\"!<u>,<\"!e,'e>}},{{{<,o,!>},<eeei!!!>i,!!!>u!!\">},{},{{<}o'<,{!!a!!!>}{{!>,<e<!>,<!!!!,>}}},{{{{<!!a!!ou!!,e!!e!!!}!!a'!!!>ioi{!>},<a!!e!!>},{{<!!o>}}},{<i!>o,'!!!>!>},<i,\"\"a,ie!>},<u\"!!'>,{}},{{{<!!!>'\"!>!>!!!>!>,<!>auu}u\"o!>},<!!'!>,<>,{<{!>,<!!!>,<a\"ioa}!>,<i!<>}}}}},{{<<!>!!!>!!\"!>,<{!>e<e{>,<!!!>,<{>},{{<\"a\"\"\",!!e!!!>!!{!>!<\"!>,<!!,u'!>!>,<>},<u<o!!ee!!e\"o>}},{<ua!>,<!>},<{!!'<!!!>!>},<>}},{}}},{{{<!>},<\"!>},<<!o,!{'!!o{!!!>a',!!i!!o!>,<o!>},<>}},{<ei!}!>},<ea{ooe',>,<!>},<!>,<!!!><>}}},{{<o{!!!!!>!>},<<'!>>,{{}}},{{},{<<\"oo!>!!!!!!'!!!!!!!>i!!!!!><\"\"u>}}},{{{{<!!,!i!>}u!!,<u,!>},<!>,<a!!{!!<uo>},<!>{!!!!!>!'{\"u!!!>!!!>!>},<!!>}},{{{{{},{<!>},<u!!i<a>,{{{<e!!o!,'u!!>,{{{<u!!e!!!{!!!>,<,<o>}},{<!>,<{<!!!>ei!!!!!>!>},<a,,>,{}}}},{}}}}},{{}},{}},{{<}\">}}},{{<!!{u!i!a>},{},{}}},{{{{{<a!!!>},<u\"!!!>},<o!>!!!>u}a<>},{<!><u<!!!!!>,<!!!>eu!!!!!>!}o!!>}}},{{},<,!<!!!>,<!>},<!!'\"!!!>!!'>},{{{<o!>,<!!}u!!!>!!\">}},{<<!!{'!>},<<!!e<!!'!!!i!!!>,<>}}},{}},{}}},{{<u!>,<!!!!!!!>},<}{,!!'}\"!>,<!>,<\"{!!<iu>,{<i!>e}oi!e!!!>!>!>>}},{{<<a'{i!>,<!}!>,<!!!>'}}}!>'a}!>},<!>>}}}},{{{},{{{<!>!!ae<!!'!>\"!>},<!!!!{!!!!!i>},<!!i>},{{{{<!!{a,!<i{o!!o!!a!>},<u'u!!{>}},{{{<,u!>\"!a!'!>},<>,{<!}ee!\"e!{!!!>,<e>}},<!!,!!{a!!!>!!!o!'\",>},{{{<!>},<!>!>},<!!!>!!<i>,{}},{<a,'<{!}},!!i{<>}},<!>,<<!!u>}}},{<\"!o!!!!!>i<<!>>,<a,\"!!!e>},{{<{!!!>a!'uo!!{o!\"u,!!,!>,<a!u!!eoe>},{{{},{<<!!!>i!!e!>},<\"!>},<ao{!>!!!!,!!!>>}},{{<e!!'{u!>,<a!\"!,u>},{{<!!'!>\"!!!!!>}<{!!!>>},{<a!!!>e!!!>!{!<!!!>},<,'<u!!!>},<e>}}},{}},{{<!>,<!o}>},<''<!!!>,<!<!!'!!\"!>,<!!a!!,>}}},{}}},{{<!,i!!!>ee{!!,u!!i<!>},<!>,<!>},<,!>!!!!}>,{<\"!!a\">}}},{{{{<!!!>oeeau!>,<,i<!>},<,ae!>!!>},{<>}}}},{{},{{{<{,\",a'{a!!u!>},<,!!!!e!!!>>}}}}}},{{{{{{<!!!!!!i!!!><!>!!\">},{<!!!>aoa'{{ea!!\"!!!>!>},<!!!>>}},{<u{\"'<!'>},{{<!i!!e!>,<!}\"a<>}}},{{<!''u!!!!o}!,!>a!!!a!>,<!>,<>,<!>},<}u!>},<u!<\"!!!>{'!>,<{>}}},{{<!!aa!!!>\"!!!>,<'!i,}\"!!!>>},{{{<u>}},{{<e!>},<ui\"!>,<!>a!!!!!!,>},{{{<!!!>!!!!e<!>},<>}}}}},{{{{},<>},{{<}!!!>},<'u!>ue!>,<!!!>!!a!ia\"a,ua>,<i!e>},{<>}},{<,<!>!!\"!>},<!!!>!!ai!e\"!>{o>,{<!>,<!>},<!>!>,<}oe'i,!>,<>}}},{{<!>,<a!>,<>},<!!e}!>!>},<!>},<io!>,<!!!>!!!>!!eoe!>},<>}}},{{{{{{<!>,<,!><,!!!>,<a>,<!!!!!>}}oo\"{!!!>!>},<!!!>}!!ae>}},{{{}},{{{{<!>,<i<!<e!>},<!>!>},<!!!!!!!>!>},<!!!i\"!!o!{i!>,<i}>},<'!!!>},<!!!>,<\"!!,{!>},<e{>}}},{{{<}!!!>},<!!'!>},<!a!>!!!!!>auo!!u'o!!!>>},{<!!e!>,,,!!!!!!!>!o!!\"u!>},<!!e!u!>},<>}}}}}},{<!!!oe!>}!!\">,<!>},<!!!>}!!!>,<>}},{{<{!>},<!>},<!>},<}!>,<!!,!}!\"\",!!!>!\"!!!>'!>,<!!{>},{<!!\"a!a!a!!!>{!><u!!!>!>},<eau!>},<\"<ii>}}}},{{{{<!>},<ei!!!>!uui!!a,!!}!>,<>},{}},{{{<u!>oa,}!>!>,<!>},<\">}},{<!!!!!>!!'!!a>}}},{{{{{<uo!!!>u'!!u!>},<}>}},{<!!!>\"!!!>o\"!!i{}!!>}},{{{}},<,o}!><!ua!>},<>},{<,>,<\"e!!i}!!!>,<!>,<!>,<>}},{<!!a!>},<>,{}},{<!!e!>'}!>},<!ii!!!>!>,<!>},<!>},<!!>,<aa,iui{!!u!'>}},{{<\"a!>},<!!}!>,>,{}},{{},<!!<}e!!!>,<!>,<!>'{e<>},{<!!!>},<!>,<eo\"!!!>u!>},<!!!>!!,\"!!!>},<!>,<!>,<!!{i!!e>,<o<!<!>},<!>},<!>>}}},{{{{<a!>},<'!>,<i!>},<,!!{<a!>,<,>,{<a,<!!!>i!!!>!o!!!!!>,<!>,<!>!>,<}!>\">}},{<}!>,<!!u!!ae>,{<}!!!>!!!!!>!!!!!>},<!!o}e>}}},{{{<oi!!!!e!>,<!!!!{,!>!!!}!>,<ee!!!!!>!!!>!>},<{e!>>}},{{{},<\"e,!!!>},<,ae}!>!!!>u!>'a!!!>},<,>},{{}}},{{{<!><u!>,<!>,<!!!>\"{a!!!>!!!>{<>}}}},{<!>,<<<!ei{}!>,<>,{<u,!>!,!\"i!>},<!!e!}>,<'<o'>}}},{{{{<!!'!{i>},{}},{{<!>,<e!>,<!!\",!o!>,<!>!>},<u<e!!!>},<o>,{{},{{},{{<!!!>\"e!>},<,!>,<>}}}}}},{<o<!!!>!!!<!>,<o!>,<ia<!>},<!!!>'>}},{{<'!>,<!!!a!!ee!>,<ee\",!,o{>},{}},{{<{!>,<e\"e!>,<'!!!!<i{!!!>,<,o!'e>}}},{{{},{<'!>,<!>},<}!{,!!>,{<u!!!>'\"u}!!\",'!!!>'\"i!!,>}}},{}},{{{<a!\"{!!ei!!!>i!!!{>},<!!!>>},{{{<!!!!<,,<!!!>!>u\"!>},<'!o{o!!!>>},<<!<!>!o>},{{},<!i',!>,<!<}!>,!>},<>}},{{},{{{<!}!!}ou!!>}}}}}}},{{{{<!>!<oe{uo!>!e,!!}!!!>},<e>,{}},{<!!!!!!i!>!>},<>,{<<!>!!u>}}},{{{},<a<}o!>},<!!{ui!>,<e}i,!!}<>},{<u>,<}<a!>}>},{}}},{{{{{{{<\"!!!>'}!!!>!'\"\">,{{<!>!!i,!!!\"i'>,<>},<!>},<o>}},{}}},<<!!!>>},<a!!!e{<}{uu!}!!!!'!>,<!>,<!!\"!!<!>},<>},{<<!>\"!!!>},<,>,<!>,<!!\"!!!>!>!!!>!>},<!><e'!!!>,e}>},{{{<\"ua}i!!!>,<>},{{},{}}},{{<'!>},<>,{{{},<'!i!a,<!!!,!!,<>}}},{<!!!!!>,<u!!e\"a!>,<!!!>,<'!>},<,e'<>}},{<}u!<'<'{u<',!>},<!!!!!>{!!ue>,<!>,<!!!>,}i>}}},{{{<!!!>!,o\"!>,<'\"!!!>!!!>\"!o{o}!}!'!!<!>>},{{<!!!>!!!>,!!!>a}!!{}a'u>},<ae!,o!!<o,!!io!<>}},{{{<!>,<!!!>!!!>e!!!>!>,<>}},{{<ei!>},<>}}}}},{{{{{}},{{<!!u!!!!'\"ii!!\"\"<!>},<>,<}u\"'>},{{}}}},{{{<>,<!}o!!!!!>a!!!!!!\",!!'>},{<!!!>{!o!!!!!}e}<>},{<a!>}!!!>!o},}{}'aeee!o>}},{{{<!!!!ua{!>},<e<{!!!>!!<ao>},{}},{{},{}},{{{<<}>},<\"!!,u!>},<!!!>!>},<!!!!!>\"{!\",!,!!!>},<ue>}}},{{{<!!a,'!!!>'o!!i!>},<!\"ue!!{!!!>,<>},{<e!>,<e{!!!!ue!<}!>},<\"!!!',u{!>,<,>,{<!!!>!!!>!>},<!!,>}}}},{{{},{{{<!!!>}!!<!>aou<'>}},{{}},{{{},<!>'uae,u\"i!>},<u,!!!!''!!>},{}}},{{},<a>}}}}},{{{{<!!!!o\"!>,<!>},<!a!>!u!u}o}!!!>a!<>},<uo\"i!>},<!!}<>},{{{<o<!>},<!!!>},<i!!!>!!}<e,!!!>u}>,<!!e<u{}'}!i!'!>},<<!!!>o>},{{{{<!!!>},<!>eo'!>},<'}{\"!!>},<{o!!!!!>i>}},{<!>!!!>!!!!u>,<\"oooe>},{<!>>,{}}}},{{},<e!>},<!!!>!!!>,<{u<u!{!}{>}},{}}},{{{<{{!!!>!!!!u!!'!!!>},<!>},<!>!!!!u!!a>},{<!>},<!u>}},{{<\"eu!!}!>!!!>}!>,<>,{<!!,!e!!!>!>},<ioe!!e!!!>i!><!,!>!!!>>,<!>,<!!!>>}},{{<>},{}},{{{<{!>,<!>,<!ua!>},<>},{<!!!>eo!!!>{}}{!\"i{!!!!{!!'{a!>i>}}}},{{{<{!!!>!>},<e\"}u{<u\"\"eu>,<{!>,<o\"!>u>}}}},{{<i!>ou<<<!!!!!>,<!!!!!>!>},<>,{<!!>}},{{{<o!>},<,ai\"!!!>{!!,!!{u!>,<o!>!>},<!!!>,<>,{{<\"!!!>},<!!ue!>!!!'!!!>\"u>}}},{{}},{<!!u'e<!!<u<<!>},<\"ui}e!!\"!>>}},{{<}a>}},{{<,\"!>!!!>\"}>},<i!>},<!><'\"e<!\"e,!}\"{',\"'\">}}}},{{{},{{},{<,{{>}}},{{{<{!!}aua\"\"o!!a!!eoe,{>,<ii!!!!!!!e!!<u!!u!!\"!!!>!>},<!>au>}},{{},{{<a!!!>},<!!!!aou'>}},{<\",e>,{<,!!}ou\"'!!!>!!!!ooe>}}}}}},{{{{<!!!!o!><o!!{!'!>,<\"!!!>{}u!>!\",!>>,{<!!'a!\",>}},{{<!!!>!}}!!o!!a!!!>},<!e!!!><o!><>},{<!>,<!u>}}},{{{<o!o!>,<!{!!!>},<e\"!!!>a}!!{{'>},<!!!>},<!>!!!>o!'>},{{<u!!!>},<e!!!>{{\"!>,<i!>,<!!!!!><'!>},<!!!>>}}}},{{{{}},{<!>ooo<!>,<ee<i''>,<u\"<e}<!>''!a{!!!>a}!>,<\">},{}}},{{},{{{<i!!!>e!>},<}<!><!!{o!>},<,!o!>},<>},<i{u{o!!<e!!!e!!i!!!>e,ua!>},<{>}}}}}";

console.log(groupSum(day9input));
